core/org.wso2.carbon.tomcat.ext/src/main/java/org/wso2/carbon/tomcat/ext/transport/CarbonResponseWrapper.java

```diff
@@ -72,6 +86,15 @@ public String generateCookieString(Cookie cookie) {
         if (super.request == null) {
             super.request = this.request;
         }
+
+        if (StringUtils.isNotBlank(legacyUserAgentRegex) && cookie instanceof ServletCookie && isLegacyUserAgent() &&
+                SameSiteCookie.NONE.getName().equals(((ServletCookie) cookie).getSameSite().getName())) {
```

Comment:
`((ServletCookie) cookie).getSameSite().getName() ` is this null safe?

Reply:
fixed. with d15ca12d142ce4219ff2380eaa7a9926a3ca142d